experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { Any, SchemaBuilder } from "../../../feature-libraries";
+import { FieldAnchor, FieldKey, TreeNavigationResult, UpPath } from "../../../core";
+import { forestWithContent } from "../../utils";
+import { brand } from "../../../util";
+import {
+	LazyOptionalField,
+	LazySequence,
+	LazyValueField,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/editable-tree-2/lazyField";
+import { getReadonlyContext } from "./utils";
+
+const detachedField: FieldKey = brand("detached");
+const detachedFieldAnchor = { parent: undefined, fieldKey: detachedField };
+
+describe("lazyField", () => {
+	it("Querying the path for editing throws for detached trees", () => {
+		const builder = new SchemaBuilder("lazyTree");
+		builder.struct("empty", {});
+		const schema = builder.intoDocumentSchema(SchemaBuilder.fieldOptional(Any));
+		const forest = forestWithContent({ schema, initialTree: {} });
+		const context = getReadonlyContext(forest, schema);
+		const cursor = context.forest.allocateCursor();
+		assert.equal(
+			forest.tryMoveCursorToField({ fieldKey: detachedField, parent: undefined }, cursor),
+			TreeNavigationResult.Ok,
+		);
+		const parentNodePath: UpPath = {
+			parent: undefined,
+			parentField: detachedField,
+			parentIndex: 0,
+		};
+		const parentAnchor = forest.anchors.track(parentNodePath);
+		const nestedFieldAnchor: FieldAnchor = {
+			parent: parentAnchor,
+			fieldKey: brand("nested"),
+		};
+		const fields = [
+			new LazySequence(
+				context,
+				SchemaBuilder.fieldSequence(Any),
+				cursor,
+				detachedFieldAnchor,
+			),
+			new LazySequence(context, SchemaBuilder.fieldSequence(Any), cursor, nestedFieldAnchor),
+			new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(Any),
+				cursor,
+				detachedFieldAnchor,
+			),
+			new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(Any),
+				cursor,
+				nestedFieldAnchor,
+			),
+			new LazyValueField(context, SchemaBuilder.fieldValue(Any), cursor, detachedFieldAnchor),
+			new LazyValueField(context, SchemaBuilder.fieldValue(Any), cursor, nestedFieldAnchor),
+		];
+		cursor.free();
+		for (const field of fields) {
+			assert.throws(
+				() => field.getFieldPathForEditing(),
```

Comment:
since you are only testing the method from LazyField, not any sub class specific behavior, I think testing one field would be sufficient. You can make an empty test subclass of LazyField (to work around it being abstract: since its constructor is public you don't even need to forward that) and just construct a single instance.

Then I'd recommend picking between one of:
1. Only test the root cases: test coverage for treeStatus should take care of root vs non root issues with the assert. This menas the test will make a single call to getFieldPathForEditing, checking it throws, and maybe one call testing tht it doesn't throw.
2. Keep root and non root test cases, and rename this test to "getFieldPathForEditing" and check that the non-erroring case is correct as well: in this case testing non-root paths also kinda makes sense.

Additionally, it might be nice to generalize this test to a getFieldPathForEditing test

Reply:
I would say that I am also testing that the concrete LazyField subclasses call `getFieldPathForEditing`. If I made the changes you describe then it would be easy for, say, `LazySequence` to call `getFieldPath` insted of calling `getFieldPathForEditing`.

I agree that the coverage that this test add is a bit odd. I will keep only the root cases.

Reply:
Or rather, that's what I aimed to cover but didn't.

Reply:
Ok, the test now acutally tests that subclasses.